hawc/apps/summary/filterset.py

```diff
@@ -0,0 +1,147 @@
+import django_filters as df
+from django.db.models import Q
+
+from ..common.filterset import BaseFilterSet
+from . import constants, models
+
+
+class VisualFilterSet(BaseFilterSet):
+    title = df.CharFilter(field_name="title", lookup_expr="icontains", label="Title text")
+    type = df.ChoiceFilter(
+        field_name="visual_type",
+        label="Visualization type",
+        help_text="Type of visualization to display",
+        empty_label="<All>",
+    )
+    published = df.ChoiceFilter(
+        choices=[(True, "Published only"), (False, "Unpublished only")],
+        label="Published",
+        help_text="Published status for HAWC visualization",
+        empty_label="<All>",
+    )
+
+    class Meta:
+        model = models.Visual
+        fields = ["title", "type", "published"]
+        grid_layout = {
+            "rows": [
+                {"columns": [{"width": 6}, {"width": 3}, {"width": 3}]},
+            ]
+        }
+
+    def filter_queryset(self, queryset):
+        queryset = super().filter_queryset(queryset)
+        query = Q(assessment=self.assessment)
+        if not self.perms["edit"]:
+            query &= Q(published=True)
+        return queryset.filter(query)
+
+    def get_type_choices(self):
```

Comment:
I moved this into it's own function so it's simpler to override, and makes the `super()` calls simpler